tests/backends/data/test_fs.py

```diff
@@ -0,0 +1,979 @@
+        list(backend.list(target))
+
+
+@pytest.mark.parametrize(
+    "containers,target,expected",
+    [
+        # Given an empty default directory, the `list` method should yield nothing.
+        ([], None, []),
+        # Given a default directory containing one file, the `list` method should yield
+        # the absolute path of the file.
+        (["foo/file_1"], None, ["/foo/file_1"]),
+        # Given a relative `target` directory containing one file, the `list` method
+        # should yield the absolute path of the file.
+        (["/foo/bar/file_1"], "bar", ["/foo/bar/file_1"]),
+        # Given a default directory containing two files, the `list` method should yield
+        # the absolute paths of the files.
+        (["foo/file_1", "foo/file_2"], None, ["/foo/file_1", "/foo/file_2"]),
+        # Given a `target` directory containing two files, the `list` method should
+        # yield the absolute paths of the files.
+        (["bar/file_1", "bar/file_2"], "/bar", ["/bar/file_1", "/bar/file_2"]),
+    ],
+)
+def test_backends_data_fs_data_backend_list_method_without_history(
+    containers, target, expected, fs_backend, fs
+):
+    """Tests the `FSDataBackend.list` method without history."""
+    # pylint: disable=invalid-name
+    for container in containers:
+        fs.create_file(container)
+
+    backend = fs_backend()
+    result = backend.list(target)
+    assert isinstance(result, Iterable)
+    assert sorted(result) == expected
+
+
+@pytest.mark.parametrize(
+    "containers,target,expected",
+    [
+        # Given an empty default directory, the `list` method should yield nothing.
+        ([], None, []),
+        # Given a default directory containing one file, the `list` method should yield
+        # a dictionary containing the absolute path of the file.
+        (["foo/file_1"], None, ["/foo/file_1"]),
+        # Given a relative `target` directory containing one file, the `list` method
+        # should yield a dictionary containing the absolute path of the file.
+        (["/foo/bar/file_1"], "bar", ["/foo/bar/file_1"]),
+        # Given a default directory containing two files, the `list` method should yield
+        # dictionaries containing the absolute paths of the files.
+        (["foo/file_1", "foo/file_2"], None, ["/foo/file_1", "/foo/file_2"]),
+        # Given a `target` directory containing two files, the `list` method should
+        # yield dictionaries containing the absolute paths of the files.
+        (["bar/file_1", "bar/file_2"], "/bar", ["/bar/file_1", "/bar/file_2"]),
+    ],
+)
+def test_backends_data_fs_data_backend_list_method_with_details(
+    containers, target, expected, fs_backend, fs
+):
+    """Tests the `FSDataBackend.list` method with `details` set to `True`."""
+    # pylint: disable=invalid-name,too-many-arguments
+    for container in containers:
+        fs.create_file(container)
+        os.utime(container, (1, 1))
```

Comment:
After reading your tests, I am not sure container is an appropriate semantic for the FS backend. Should we be more specific, _e.g._ use `files` instead?

Reply:
Well spotted) The idea was to use common semantics across data backend tests, however, indeed, this makes the test less obvious( Updated this part)